File: flask-mvc-starter-kit/app/models/Submission.py
from flask_sqlalchemy import SQLAlchemy
from app import db

class Submission(db.Model):
    __tablename__ = 'Submission'
    
    submissionId = db.Column(db.Integer, primary_key=True, autoincrement=True)
    assignmentId = db.Column(db.Integer, db.ForeignKey('Assignment.assignmentId', ondelete="CASCADE"), nullable=False)
    id_student = db.Column(db.String(50), db.ForeignKey('User.id_user', ondelete="CASCADE"), nullable=False)
    submission_date = db.Column(db.DateTime, default=db.func.current_timestamp())
    file_path = db.Column(db.String(255))
    grade = db.Column(db.Numeric(5, 2))
    comments = db.Column(db.Text)
    taskReviewed = db.Column(db.Boolean, default=False)
    taskSubmitted = db.Column(db.Boolean, default=False)

    # Relaciones
    assignment = db.relationship("Assignment", backref=db.backref("submissions", lazy=True))
    student = db.relationship("User", backref=db.backref("submissions", lazy=True))

    def to_dict(self):
        return {
            "submissionId": self.submissionId,
            "assignmentId": self.assignmentId,
            "id_student": self.id_student,
            "submission_date": self.submission_date,
            "file_path": self.file_path,
            "grade": str(self.grade),
            "comments": self.comments,
            "taskReviewed": self.taskReviewed,
            "taskSubmitted": self.taskSubmitted,
            "assignment": self.assignment.to_dict() if self.assignment else None,
            "student": self.student.to_dict() if self.student else None
        }
    
    

File: src/main/java/Model/Submission.java
package Model;
import com.google.protobuf.Timestamp;
import java.math.BigDecimal;

/**
 * Representación de la tabla 'Submission' en la base de datos.
 * 
 * CREATE TABLE Submission (
 *  submissionId INT AUTO_INCREMENT PRIMARY KEY,
 *  assignmentId INT,                            
 *  id_student VARCHAR(50),                               
 *  submission_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP,  
 *  file_path VARCHAR(255),                       
 *  grade DECIMAL(5,2),                          
 *  comments TEXT,                                
 *  taskReviewed BOOLEAN DEFAULT FALSE,          
 *  taskSubmitted BOOLEAN DEFAULT false,          
 *  FOREIGN KEY (assignmentId) REFERENCES Assignment(assignmentId) ON DELETE CASCADE,
 *  FOREIGN KEY (id_student) REFERENCES User(id_user) ON DELETE CASCADE
 * );
 * 
 * @author kathe
 */
public class Submission {
    private int submissionId;
    private Assignment assignmentId;
    private User id_student;
    private Timestamp submission_date;
    private String file_path;
    private BigDecimal grade; 
    private String comments;
    private boolean taskReviewed;
    private boolean taskSubmitted;

    // Constructor
    public Submission(int submissionId, Assignment assignmentId, User id_student, Timestamp submission_date,
                         String file_path, BigDecimal grade, String comments, boolean taskReviewed, boolean taskSubmitted) {
        this.submissionId = submissionId;
        this.assignmentId = assignmentId;
        this.id_student = id_student;
        this.submission_date = submission_date;
        this.file_path = file_path;
        this.grade = grade;
        this.comments = comments;
        this.taskReviewed = taskReviewed;
        this.taskSubmitted = taskSubmitted;
    }

    public int getSubmissionId() {
        return submissionId;
    }

    public void setSubmissionId(int submissionId) {
        this.submissionId = submissionId;
    }

    public Assignment getAssignmentId() {
        return assignmentId;
    }

    public void setAssignmentId(Assignment assignmentId) {
        this.assignmentId = assignmentId;
    }

    public User getIdStudent() {
        return id_student;
    }

    public void setIdStudent(User id_student) {
        this.id_student = id_student;
    }

    public Timestamp getSubmissionDate() {
        return submission_date;
    }

    public void setSubmissionDate(Timestamp submission_date) {
        this.submission_date = submission_date;
    }

    public String getFilePath() {
        return file_path;
    }

    public void setFilePath(String file_path) {
        this.file_path = file_path;
    }

    public BigDecimal getGrade() {
        return grade;
    }

    public void setGrade(BigDecimal grade) {
        this.grade = grade;
    }

    public String getComments() {
        return comments;
    }

    public void setComments(String comments) {
        this.comments = comments;
    }

    public boolean isTaskReviewed() {
        return taskReviewed;
    }

    public void setTaskReviewed(boolean taskReviewed) {
        this.taskReviewed = taskReviewed;
    }

    public boolean isTaskSubmitted() {
        return taskSubmitted;
    }

    public void setTaskSubmitted(boolean taskSubmitted) {
        this.taskSubmitted = taskSubmitted;
    }

    // Método toString para representar la información de la tarea
    @Override
    public String toString() {
        return "SubmissionAPI{" +
                "submissionId=" + submissionId +
                ", assignmentId=" + assignmentId +
                ", id_student=" + id_student +
                ", submission_date=" + submission_date +
                ", file_path='" + file_path + '\'' +
                ", grade=" + grade +
                ", comments='" + comments + '\'' +
                ", taskReviewed=" + taskReviewed +
                ", taskSubmitted=" + taskSubmitted +
                '}';
    }
}
